src/Microsoft.IdentityModel.Protocols.SignedHttpRequest/SignedHttpRequestValidationParameters.cs

```diff
@@ -0,0 +1,229 @@
+    /// <param name="signedHttpRequest">A SignedHttpRequest.</param>
+    /// <param name="signedHttpRequestValidationContext">A structure that wraps parameters needed for SignedHttpRequest validation.</param>
+    /// <param name="cancellationToken">Propagates notification that operations should be canceled.</param>
+    /// <returns>A <see cref="SecurityKey"/> used to validate a signature of the <paramref name="signedHttpRequest"/>, otherwise expected to throw an appropriate exception.</returns>
+    public delegate Task<SecurityKey> SignedHttpRequestSignatureValidatorAsync(IEnumerable<SecurityKey> popKeys, SecurityToken signedHttpRequest, SignedHttpRequestValidationContext signedHttpRequestValidationContext, CancellationToken cancellationToken);
+
+    /// <summary>
+    /// Defines a set of parameters that are used by a <see cref="SignedHttpRequestHandler"/> when validating a SignedHttpRequest.
+    /// </summary>
+    public class SignedHttpRequestValidationParameters
+    {
+        private TimeSpan _signedHttpRequestLifetime = DefaultSignedHttpRequestLifetime;
+
+        /// <summary>
+        /// Gets or sets a value indicating whether the unsigned query parameters are accepted or not.
+        /// </summary>
+        /// <remarks>https://tools.ietf.org/html/draft-ietf-oauth-signed-http-request-03#section-5.1</remarks>
+        public bool AcceptUnsignedQueryParameters { get; set; } = true;
+
+        /// <summary>
+        /// Gets or sets a value indicating whether the unsigned headers are accepted or not. 
+        /// </summary>
+        /// <remarks>https://tools.ietf.org/html/draft-ietf-oauth-signed-http-request-03#section-5.1</remarks>
+        public bool AcceptUnsignedHeaders { get; set; } = true;
+
+        /// <summary>
+        /// Gets or sets the <see cref="AdditionalClaimValidatorAsync"/> delegate.
+        /// </summary>
+        public AdditionalClaimValidatorAsync AdditionalClaimValidatorAsync { get; set; }
+
+        /// <summary>
+        /// Gets or sets a collection of <see cref="SecurityKey"/> used for the 'cnf' claim decryption.
+        /// </summary>
+        public IEnumerable<SecurityKey> CnfDecryptionKeys { get; set; }
+
+        /// <summary>
+        /// Gets or sets the <see cref="SignedHttpRequestSignatureValidatorAsync"/> delegate.
+        /// </summary>
+        public CnfDecryptionKeysResolverAsync CnfDecryptionKeysResolverAsync { get; set; }
+
+        /// <summary>
+        /// Default value for the <see cref="SignedHttpRequestLifetime"/>.
+        /// </summary>
+        public static readonly TimeSpan DefaultSignedHttpRequestLifetime = TimeSpan.FromMinutes(5);
+
+        /// <summary>
+        /// Gets or sets a custom HttpClient when obtaining a JWK set using the 'jku' claim.
+        /// </summary>
+        /// <remarks>https://tools.ietf.org/html/rfc7800#section-3.5</remarks> s
+        public HttpClient HttpClientForJkuResourceRetrieval { get; set; }
```

Comment:
Could this be HttpClientFactory for Core?

Reply:
I'm not sure how to implement that. We don't have a Core-only target.

Reply:
todo: Replace with a delegate.